frappe/templates/includes/comments/comments.py

```diff
@@ -54,9 +54,12 @@ def add_comment(comment, comment_email, comment_by, reference_doctype, reference
 		pass
 	else:
 		# notify creator
+		creator_email = frappe.db.get_value("User", doc.owner, "email") or doc.owner
+		subject = _("New Comment on {0}: {1}").format(doc.doctype, doc.get_title())
```

Comment:
If there's no title field this will return empty string!

Add name as fallback here.